server/index.js

```diff
@@ -64,59 +100,102 @@ app.get('/tetraflix/genre/:genre', (req, res) => {
     });
 });
 
-app.post('/tetraflix/sessionData', (req, res) => {
-  const { events } = req.body;
+const receiveSessionData = () => {
+  let deleteId;
+  let user;
   const movies = [];
-  events.forEach((event) => {
-    movies.push([event.movie.id, event.progress]);
-    if (event.progress === 1) {
-      postgresDb.Movie.increment('views', { where: { id: event.movie.id } })
-        .catch((err) => {
-          throw err;
+  const sessionDataOptions = {
+    QueueUrl: queues.sessionData,
+  };
+  receiveMessages(sessionDataOptions)
+    .then((data) => {
+      if (!data || !data.Messages) {
+        throw new Error('No messages to receive');
+      } else {
+        const message = JSON.parse(data.Messages[0].Body);
+        const { events } = message;
+        user = message.userId;
+        deleteId = data.Messages[0].ReceiptHandle;
+        events.forEach((event) => {
+          movies.push([event.movie.id, event.progress]);
+          if (event.progress === 1) {
+            postgresDb.Movie.increment('views', { where: { id: event.movie.id } })
+              .catch((err) => {
+                throw err;
```

Comment:
Make sure that inside of your catch blocks, you're not throwing another error (since the catch block is meant to catch any errors thrown in the corresponding promise chain). You could refactor to use console.logs instead!